rfcs/2023-05-03-data-volume-metrics.md

```diff
@@ -0,0 +1,146 @@
+# RFC 2023-05-02 - Data Volume Insights metrics
+
+Vector needs to be able to emit accurate metrics that can be usefully queried
+to give users insights into the volume of data moving through the system.
+
+## Scope
+
+### In scope
+
+- All event metrics within Vector need to emit the estimated JSON size of the
+  event. With a consistent method for determining the size it will be easier
+  to accurately compare data in vs data out.
+- The metrics sent by each sink needs to be tagged with the source id of the
+  event so the route an event takes through Vector can be queried.
+- Each event needs to be labelled with a `service`. This is a new concept
+  within Vector and represents the application that generated the log or
+  metric.
+- The service tag and source tag in the metrics needs to be opt in so customers
+  that don't need the increased cardinality are unaffected.
+
+### Out of scope
+
+- A separate metric, `component_sent_bytes_total` that indicates network bytes
+  sent by Vector is not considered here.
+
+## Pain
+
+Currently it is difficult to accurately gauge the volume of data that is moving
+through Vector. It is difficult to query where data being sent out has come
+from.
+
+## Proposal
+
+### User Experience
+
+Global config options will be provided allowing the `service` tag and the
+`source` tag to be specified. For example:
+
+```yaml
+metrics:
+  tags:
+    service: service
+    source: input
```

Comment:
I assume this is a map of where to find the field? Should we use `log_schema` instead for the mapping? Also, how does this overlap with "semantic meaning" and the associated VRL functions?

Reply:
No. This is the name of the tag that will be attached to the metric emitted by Vector. I'll make this more clear in the RFC.